src/main/kotlin/org/eclipse/tractusx/bpdmcertificatemanagement/entity/CertificateDB.kt

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2023 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.eclipse.tractusx.bpdmcertificatemanagement.entity
+
+import jakarta.persistence.*
+import java.time.ZonedDateTime
+import java.util.SortedSet
+
+@Entity
+@Table(name = "certificate")
+class CertificateDB(
+
+    @Column(name = "business_partner_number")
+    var businessPartnerNumber: String?,
+
+    @ManyToOne
+    @JoinColumn(name = "certificate_type_id", nullable = false)
+    var type: CertificateTypeDB,
+
+    @Column(name = "registration_number")
+    var registrationNumber: String?,
+
+    @Column(name = "area_of_application")
+    var areaOfApplication: String?,
+
+    @Column(name = "remark")
+    var remark: String?,
+
+    @ElementCollection(fetch = FetchType.EAGER)
+    @CollectionTable(name = "enclosed_sites", joinColumns = [JoinColumn(name = "certificate_id")])
+    var enclosedSites: SortedSet<EnclosedSiteDB>? = sortedSetOf(),
```

Comment:
When it comes to collections I would actually keep the field val but use a mutable collection. JPA (or hibernate not sure) will complain if they need to keep track of elements in the set if you assign the whole collection to something else. The framework rather wants you to change the existing collection.

I think when you want to make an update you will run into problems here

Reply:
Adapted